purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt

```diff
@@ -41,52 +43,60 @@ data class PaywallData(
     /**
      * Returns the [LocalizedConfiguration] to be used based on the current locale.
      */
-    val localizedConfiguration: LocalizedConfiguration
+    val localizedConfiguration: Pair<Locale, LocalizedConfiguration>
+        @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
         get() {
             val preferredLocales = LocaleListCompat.getDefault()
 
             for (i in 0 until preferredLocales.size()) {
                 preferredLocales.get(i)?.let { locale ->
-                    configForLocale(locale)?.let { localizedConfiguration ->
-                        return localizedConfiguration
+                    // This is a hack to make android studio previews work. In previews, the locale language method
+                    // gives a different result than expected (for example, "es-es" instead of "es").
+                    // In order to fix that, we convert the string to a locale and we parse that as a locale so the
+                    // language and country are correctly parsed.
+                    val localeToCheck = Locale.forLanguageTag(locale.toString())
```

Comment:
I would extract this to a `Locale` extension and add tests specifically for it.
This is solving a very specific problem (looking up a locale with `es-es`, for example), and putting the logic here seems like the wrong place for it.

Reply:
This is tricky... This is part of the `purchases` module, so we would need to add the extensions there. However, `forLanguageTag` uses `API 21+`. Having said that, I think it's ok to fallback to the original locale in this case, since this is mostly done to fix previews.

Reply:
Done in https://github.com/RevenueCat/purchases-android/pull/1267/commits/d01bb5546931c21d7bf4a0bd26d48a44270031ee